### PR TITLE
New Module - eks_nodegroup

### PR DESCRIPTION
**Pull request by tjarra**
##### SUMMARY
Add a new module to manage nodegroups in EKS clusters.
 
##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
eks_nodegroup

##### ADDITIONAL INFORMATION
TODO:

- [x] PR [ansible-collections/amazon.aws/pull/975](https://github.com/ansible-collections/amazon.aws/pull/975) to add waiters for EKS nodegroup.
- [x] PR [aws-terminator/pull/223](https://github.com/mattclay/aws-terminator/pull/223) to add permissions to execute integration tests.

```
- name: create nodegroup
  community.aws.eks_nodegroup:
    name: test_nodegroup
    state: present
    cluster_name: test_cluster
    node_role: arn:aws:eks:us-east-1:1231231123:role/asdf
    subnets:
      - subnet-qwerty123
      - subnet-asdfg456
    scaling_config:
      - min_size: 1
      - max_size: 2
      - desired_size: 1
    disk_size: 20
    instance_types: 't3.micro'
    ami_type: 'AL2_x86_64'
    labels:
      - 'teste': 'test'
    taints:
      - key: 'test'
        value: 'test'
        effect: 'NO_SCHEDULE'
    capacity_type: 'on_demand'
```

### CODE REVIEW

**Comment by gravesm** [comment]
The permissions for CI should now be set, but I had a difficult time getting this test suite to pass. It just takes too long. My suggestion would be to try being more economical in what is being tested if possible, and only use `wait: True` in cases where it is absolutely necessary. Also, this PR will need an alias file for the test target in order for the tests to run.

File: plugins/modules/eks_nodegroup.py [hunk 1]
```diff
@@ -0,0 +1,707 @@
...
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: eks_nodegroup
+version_added: 4.0.0
+short_description: Manage EKS Nodegroup module
+description:
+  - Manage EKS Nodegroup
```
Comment (alinabuzachis) [inline plugins/modules/eks_nodegroup.py:15]:
```suggestion
  - Manage EKS Nodegroup.
```

**Review by alinabuzachis — APPROVED** [review]

File: plugins/modules/eks_nodegroup.py [hunk 2]
```diff
@@ -0,0 +1,707 @@
...
+        description: The effect of the taint.
+        type: str
+        choices:
+          - NO_SCHEDULE
+          - NO_EXECUTE
+          - PREFER_NO_SCHEDULE
+  launch_template:
+    description:
+      - An object representing a node group's launch template specification.
+      - If specified, then do not specify instanceTypes , diskSize , or remoteAccess.
```
Comment (alinabuzachis) [inline plugins/modules/eks_nodegroup.py:121]:
```suggestion
      - If specified, then do not specify I(instance_types), I(disk_size), or I(remote_access).
```

File: plugins/modules/eks_nodegroup.py [hunk 3]
```diff
@@ -0,0 +1,707 @@
...
+    description:
+      - Purge existing tags that are not found in the nodegroup.
+    type: bool
+    default: true
+  wait:
+    description: Specifies whether the module waits until the profile is created or deleted before moving on.
+    type: bool
+    default: false
+  wait_timeout:
+    description: The duration in seconds to wait for the nodegroup to become active. Defaults to 1200 seconds.
```
Comment (alinabuzachis) [inline plugins/modules/eks_nodegroup.py:166]:
```suggestion
    description: The duration in seconds to wait for the nodegroup to become active. Defaults to C(1200) seconds.
```

File: plugins/modules/eks_nodegroup.py [hunk 4]
```diff
@@ -0,0 +1,707 @@
...
+      - AL2_x86_64
+      - AL2_x86_64_GPU
+      - AL2_ARM_64
+      - CUSTOM
+      - BOTTLEROCKET_ARM_64
+      - BOTTLEROCKET_x86_64
+  remote_access:
+    description:
+      - The remote access (SSH) configuration to use with your node group.
+        If you specify launch_template , then don't specify remote_access , or the node group deployment will fail.
```
Comment (alinabuzachis) [inline plugins/modules/eks_nodegroup.py:75]:
```suggestion
        If you specify I(launch_template) , then don't specify I(remote_access), or the node group deployment will fail.
```

File: plugins/modules/eks_nodegroup.py [hunk 5]
```diff
@@ -0,0 +1,707 @@
...
+      max_size:
+        description: The maximum number of nodes that the managed node group can scale out to.
+        type: int
+      desired_size:
+        description: The current number of nodes that the managed node group should maintain.
+        type: int
+  disk_size:
+    description:
+      - Size of disk in nodegroup nodes.
+        If you specify launch_template, then don't specify disk_size, or the node group deployment will fail.
```
Comment (alinabuzachis) [inline plugins/modules/eks_nodegroup.py:54]:
```suggestion
        If you specify I(launch_template), then don't specify I(disk_size), or the node group deployment will fail.
```

**Comment by filipecifali** [comment]
This is super useful, I'm a bit surprised it didn't exist before. Is there anything we can do to help get this merged / released?

File: plugins/modules/eks_nodegroup.py [hunk 6]
```diff
@@ -0,0 +1,707 @@
...
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: eks_nodegroup
+version_added: 5.1.0
```
Comment (tremble) [inline plugins/modules/eks_nodegroup.py:12]:
```suggestion
version_added: 5.3.0
```

**Review by tremble — APPROVED** [review]

**Comment by tremble** [comment]
Hi all,

Sorry this has taken so long to get merged.  This should be available once we release 5.3.0

@tjarra Many thanks for your work here.

**Pull request merged**